### PR TITLE
regenerate tutorials

### PR DESCRIPTION
**Pull request by steven-johnson**

### CODE REVIEW

File: tutorials/tutorial_lesson_02_input_image.html [hunk 1]
```diff
@@ -301,6 +291,46 @@
...
+ style="text-decoration:none;"><span style="width:100%; height:100%; top:0; left:0;">
+<div style="width:200px; padding:20px; background-color: #dddddd; text-align: left; ">
+<b>
+21
+&nbsp</b>
+ Auto-Scheduler
+</div></span></a>
+<div style="width:200px; height:20px;"></div>
+<a href=
+tutorial_lesson_15_generators_usage.html
```
Comment (abadams) [inline tutorials/tutorial_lesson_02_input_image.html]: It looks like these got generated in the wrong order. I think the order of arguments passed to the gen_tutorials script matters.

File: tutorials/tutorial_lesson_04_debugging_2.html [hunk 2]
```diff
@@ -343,75 +373,6 @@
...
- > Store gradient.0(1, 7) = 8
- > Store gradient.0(2, 7) = 9
- > Store gradient.0(3, 7) = 10
- > Store gradient.0(4, 7) = 11
- > Store gradient.0(5, 7) = 12
- > Store gradient.0(6, 7) = 13
- > Store gradient.0(7, 7) = 14
- > End pipeline gradient.0()
-
-</b></div>
```
Comment (abadams) [inline tutorials/tutorial_lesson_04_debugging_2.html]: Looks like these traces all disappeared, which means the snippets weren't successfully included.

Comment (steven-johnson) [reply tutorials/tutorial_lesson_04_debugging_2.html]: Where/how are the snippets produced? In Gitter you mentioned "To generate the snippets you run generate_snippets in tutorials/figures" but I don't see any scripts of any sort there -- gen_tutorials.sh appears to be the only script in the whole repo.

File: tutorials/tutorial_lesson_02_input_image.html (hunk 1 above)
Comment (steven-johnson) [reply tutorials/tutorial_lesson_02_input_image.html]: I just used `./gen_tutorials ~/my_halide_dir/tutorials/*.{cpp,sh}` -- weird that it didn't generate in the same order.

File: tutorials/tutorial_lesson_04_debugging_2.html (hunk 2 above)
Comment (abadams) [reply tutorials/tutorial_lesson_04_debugging_2.html]: tutorials/figures in the main Halide repo. The scripts are weirdly split between the two repos.

**Comment by steven-johnson** [comment]
Updated the snippets and re-pushed. Things still seem to be out-of-order, though, which makes no sense.

**Comment by steven-johnson** [comment]
Ordering issue had to do with how the files were specified to the script -- updated the comment in gen_tutorials.sh to give proper directions. PTAL.

**Comment by abadams** [comment]
Ah, I bet my shell was just doing the glob expansion differently

**Pull request merged**